File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let dp = new Array(46).fill(0);
    dp[0] = 0;
    dp[1] = 1;
    

    for(let i = 2 ; i <= 46 ; i++){
        dp[i] = dp[i-1] + dp[i-2]
    }

    return dp[n+1]
};

//아래 2층에서 올라오는 방법 
// 아래 1층에서 올라오는 방법 f(n)  = f(n-1) + f(n-2)
//memoization 사용

File: 0070-climbing-stairs/0070-climbing-stairs.go
func climbStairs(n int) int {
    dp := make([]int, n+1)
    dp[0] = 1;
    dp[1] = 1;
    

    //현재 계단을 오르는 방법은 n-1번째에서  (1칸올라오기) 올라오거나 n-2 번째에서 올라오는 방법 (2칸 올라오기)

    for i := 2 ; i <= n ; i++ {
        dp[i] = dp[i-1] + dp[i-2]
    }

    return dp[n];
}